File: app/src/main/java/com/locallinkonline/stcatherineschool/rest/api/StCatherineApi.java
package com.locallinkonline.stcatherineschool.rest.api;

import com.locallinkonline.stcatherineschool.rest.model.DataResourceResponse;
import com.locallinkonline.stcatherineschool.rest.model.LunchResponseObject;
import com.locallinkonline.stcatherineschool.rest.model.Schedule;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface StCatherineApi {

    @GET("school-schedule")
    Call<Schedule> getSchoolSchedule();

    @GET("activity")
    Call<Schedule> getActivitySchedule(@Query("activityId") String activityId);

    @GET("activity")
    Call<Schedule> getHomeworkSchedule(@Query("id") String id);

    @GET("lunch")
    Call<LunchResponseObject> getLunches(@Query("lunchDate") String date);

    @GET("data-resources")
    Call<DataResourceResponse> getDataResources();
}

File: app/src/main/java/com/locallinkonline/stcatherineschool/rest/api/StCatherineApi.kt
package com.locallinkonline.stcatherineschool.rest.api

import com.locallinkonline.stcatherineschool.rest.model.DataResourceResponse
import com.locallinkonline.stcatherineschool.rest.model.LunchResponseObject
import com.locallinkonline.stcatherineschool.rest.model.Schedule

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface StCatherineApi {

    @get:GET("school-schedule")
    val schoolSchedule: Call<Schedule>

    @get:GET("data-resources")
    val dataResources: Call<DataResourceResponse>

    @GET("activity")
    fun getActivitySchedule(@Query("activityId") activityId: String): Call<Schedule>

    @GET("activity")
    fun getHomeworkSchedule(@Query("id") id: String): Call<Schedule>

    @GET("lunch")
    fun getLunches(@Query("lunchDate") date: String): Call<LunchResponseObject>
}
